### ISSUE TITLE
json matcher selecting 0th element on null array succeeds but should fail.

### ISSUE THREAD

**Issue by davidbkemp** [etorreborre/specs2#1183]
I would expect the following spec to fail, but it passes:

```
import org.specs2.matcher.JsonMatchers
import org.specs2.mutable.Specification

class FalsePositiveSpec extends Specification with JsonMatchers {
    "this should fail" in {
      """{"items":null}""" must (/("items") /# 0 / ("id" -> "test"))
  }
}
```

Interestingly, it does fail (as expected) when you use any index other than `0`.

**Comment by etorreborre** [comment]
Hi @davidbkemp, thanks for reporting this. I think I understand where the issue is coming from. I'll try to fix it over the next few days.

**Comment by etorreborre** [comment]
@davidbkemp I fixed this issue on `main`. This is not yet backported to the `specs2-4.x` branch, which version are you using?

**Comment by davidbkemp** [comment]
Thank you very much for fixing it.  We are currently using version `4.20.0`

**Comment by etorreborre** [comment]
The fix is now published in version [`4.20.1`](https://repo1.maven.org/maven2/org/specs2/specs2-core_2.13/4.20.1/)

**Comment by etorreborre** [comment]
I think that `4.20.1` was incorrectly porting the fix. This has been corrected with `4.20.4`.
